tests/filesTest.js

```diff
@@ -1341,6 +1339,53 @@ describe('Main: Currently testing files management,', function () {
       }
       const provider = createProvider()
 
+      await provider
+        .uponReceiving(`as '${username}', a REPORT request to get files by tag`)
+        .withRequest({
+          method: 'REPORT',
+          path: MatchersV3.regex(
+            '.*\\/remote\\.php\\/dav\\/files\\/' + config.adminUsername + '\\/$',
+            '/remote.php/dav/files/' + config.adminUsername + '/'
+          ),
+          headers: {
+            ...validAuthHeaders,
+            ...applicationXmlResponseHeaders
+          },
+          body: new XmlBuilder('1.0', '', 'oc:filter-files').build(ocFilterFiles => {
+            ocFilterFiles.setAttributes({
+              'xmlns:d': 'DAV:',
+              'xmlns:oc': 'http://owncloud.org/ns'
+            })
+            ocFilterFiles.appendElement('d:prop', '', dProp => {
+              dProp.appendElement('oc:fileid', '', '')
+            }).appendElement('oc:filter-rules', '', ocFilterRules => {
+              ocFilterRules.appendElement('oc:systemtag', '', tagId)
+            })
+          })
+        })
+        .willRespondWith(getFileInfoBy('tag'))
+
+      await getCapabilitiesInteraction(provider)
+      await getCurrentUserInformationInteraction(provider)
```

Comment:
Can you move these to just after the `createProvider()`
we want to check those interactions before other interactions in a test.
(do similar for other added tests)